dismantle-tablegen/tests/Trie.hs

```diff
@@ -1,9 +1,470 @@
+{-# LANGUAGE OverloadedStrings #-}
 module Trie ( trieTests ) where
 
+import qualified Data.ByteString as BS
+import Data.Either ( fromRight )
+import           Numeric (showHex)
 import qualified Test.Tasty as T
-import qualified Test.Tasty.HUnit as T
+import           Test.Tasty.HUnit
+
+import Dismantle.Tablegen.ByteTrie ( byteTrie, lookupByte )
 
-import qualified Dismantle.Tablegen.ByteTrie as BT
 
 trieTests :: IO T.TestTree
-trieTests = return (T.testGroup "Trie" [])
+trieTests = return $
+  let hexStr v = (<>) "0x" . showHex v
+      testLookup t b expected = testCase (hexStr b " lookup")
+                                (let Right v = lookupByte t b in v @?= expected)
+  in
```

Comment:
My only suggestion is that these two definitions could be made top level to simplify the structure (removing the let)